apps/mobile/src/features/browser/approver-sheet/stx/sign-structured-message/sign-structured-message.tsx

```diff
@@ -0,0 +1,54 @@
+import { BaseStxMessageApproverLayout } from '@/features/approver/layouts/base-stx-message-approver.layout';
+import { useAccounts } from '@/store/accounts/accounts.read';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import { deserializeCV } from '@stacks/transactions';
+
+import {
+  RpcRequest,
+  RpcResponse,
+  createRpcSuccessResponse,
+  stxSignStructuredMessage,
+} from '@leather.io/rpc';
+
+interface StxSignStructuredMessageApproverProps {
+  request: RpcRequest<typeof stxSignStructuredMessage>;
+  sendResult(result: RpcResponse<typeof stxSignStructuredMessage>): void;
+  closeApprover(): void;
+  accountId: string;
+}
+
+export function StxSignStructuredMessageApprover({
+  request,
+  closeApprover,
+  sendResult,
+  accountId,
+}: StxSignStructuredMessageApproverProps) {
+  const { list: accounts } = useAccounts();
+  const signer = useStacksSigners().fromAccountId(accountId)[0];
+
+  if (!signer) throw new Error('No signer found');
+
+  async function onApprove() {
+    if (!signer) throw new Error('No signer found');
```

Comment:
Maybe we can use `StacksError` for these errors instead? Then we can have a key for the message too